app/src/pages/ExploreDataPage.tsx

```diff
@@ -13,16 +13,58 @@ import {
   PhraseSelections,
 } from "../utils/MadLibs";
 import styles from "./ExploreDataPage.module.scss";
+import CompareStatesForVariableReport from "../features/reports/CompareStatesForVariableReport";
+
+function getPhraseValue(
+  madlib: MadLib,
+  segmentIndex: number,
+  phraseSelections: PhraseSelections
+): string {
+  const segment = madlib.phrase[segmentIndex];
+  return typeof segment === "string"
+    ? segment
+    : segment[phraseSelections[segmentIndex]];
+}
+
+function ReportWrapper(props: {
+  madLibIndex: number;
+  phraseSelections: PhraseSelections;
+}) {
+  const madlib = MADLIB_LIST[props.madLibIndex];
+  switch (props.madLibIndex) {
+    case 0:
+      return (
+        <DemoReport madlib={madlib} phraseSelections={props.phraseSelections} />
+      );
+    case 1:
+      return (
+        <TellMeAboutReport
+          madlib={madlib}
+          phraseSelections={props.phraseSelections}
+        />
+      );
+    case 2:
+      return (
```

Comment:
Oh this probably way clearer in the report components haha- I can update mine to do this as well.

Reply:
This is what I meant when I was talking about converting to structured data btw. Glad you agree!